### PR TITLE
Add support to specify trusted addresses

### PR DESCRIPTION
**Pull request by HashidaTKS**
Add support to specify trusted addresses.

If a trusted domain entry contains "@", we regard it as address, not domain.

## Test

* Remove files in `%APPDATA%\FlexConfirmMail` if exist.
* Open Outlook and open setting dialog for FlexConfirmMail
* Open the "社内ドメイン" tab of the setting dialog.
  * Confirm that the comment have description for how to specify addresses like below  ...OK
    ```
    # (5) ドメインだけでなくアドレス全体を指定することもできます。
    #     * @を含む場合はアドレスが指定されたとみなします。
    #     * @を含まない場合はドメインが指定されたとみなします。
    #
    ```
* Add entries as following to the "社内ドメイン" setting
  ```
  test@a.co.jp
  *test@b.co.jp
  -*test@c.co.jp
  test2@*
  test.co.jp
  ```
* Try to send mail to `test@a.co.jp` and  `test3@a.co.jp` （test for `test@a.co.jp`）
  * Confirm that `test@a.co.jp` is in "社内の送信先" ...OK
  * Confirm that `test3@a.co.jp` is in "社外の送信先" ...OK
* Try to send mail to `test-test@a.co.jp` and  `test-test@b.co.jp` (test for `*test@b.co.jp`)
  * Confirm that `test-test@b.co.jp` is in "社内の送信先" ...OK
  * Confirm that `test-test@a.co.jp` is in "社外の送信先" ...OK
* Try to send mail to `test-test@a.co.jp` and `test-test@c.co.jp` (test for `-*test@c.co.jp`)
  * Confirm that `test-test@a.co.jp` and  `test-test@c.co.jp` are in "社外の送信先"  ...OK
* Try to send mail to `test2@a.co.jp` and `test2@c.co.jp` (test for `test2@*`)
  * Confirm that `test2@a.co.jp` and  `test2@c.co.jp` are in "社内の送信先" ...OK
  * Confirm that they are grouping each by  `a.co.jp` and `b.co.jp` ...OK
* Try to send mail to `test-test@test.co.jp` and `test-test@a.co.jp` (test for `test.co.jp`)
  * Confirm that `test-test@test.co.jp` is in "社内の送信先" ...OK
  * Confirm that `test-test@a.co.jp` is in "社外の送信先" ...OK

### CODE REVIEW

File: Dialog/MainDialog.xaml.cs [hunk 1]
```diff
@@ -136,7 +136,7 @@ private List<string> ToLower(List<string> list)
             return ret;
         }
 
-        private bool IsTrustedDomain(string domain, HashSet<string> trusted)
```
Comment (HashidaTKS) [inline Dialog/MainDialog.xaml.cs:139]: 今回の修正のついでに、引数が使用されていないので削除した。

File: Properties/Resources.Designer.cs [hunk 2]
```diff
@@ -324,8 +324,8 @@ public class Resources {
         }
         
         /// <summary>
-        ///   宛先に多数のドメインが検知されました。
-        ///ToおよびCcに含まれるメールアドレスはすべての受取人が確認できるため、
+        ///   宛先に多数のドメインが検知されました。
```
Comment (HashidaTKS) [inline Properties/Resources.Designer.cs:327]: 改行コード違いによる差分

**Comment by piroor** [comment]
I've confirmed the automated build based on this commit works as expected.

And I've realized that the full address syntax is available only for internal (trusted) domains. Patterns like `dangerous*@example.com` in unsafe domains looks to produce nothing. This is one difference from the [Thunderbird version](https://github.com/FlexConfirmMail/Thunderbird/pull/43). I think it is better to support new syntax in both domains to reduce users' confusion. How about this idea?

**Comment by HashidaTKS** [comment]
> I've confirmed the automated build based on this commit works as expected.
> 
> And I've realized that the full address syntax is available only for internal (trusted) domains. Patterns like `dangerous*@example.com` in unsafe domains look to produce nothing. This is one difference from the [Thunderbird version](https://github.com/FlexConfirmMail/Thunderbird/pull/43). I think it is better to support new syntax in both domains to reduce users' confusion. How about this idea?

Sure, I am planing to also add this feature for unsafe domains.
I have created this PR as a first job of adding support for specifying addresses.

I am creating a new PR for unsafe domains now.

**Pull request merged**